File: solutions/csharp/vendingmachine/Coin.cs
namespace VendingMachineApp
{
    public enum Coin
    {
        Penny = 1,
        Nickel = 5,
        Dime = 10,
        Quarter = 25
    }

    public static class CoinExtensions
    {
        public static double GetValue(this Coin coin)
        {
            return (int)coin / 100.0;
        }
    }
}

File: solutions/java/src/vendingmachine/Coin.java
package vendingmachine;

public enum Coin {
    PENNY(0.01),
    NICKEL(0.05),
    DIME(0.1),
    QUARTER(0.25);

    private final double value;

    Coin(double value) {
        this.value = value;
    }

    public double getValue() {
        return value;
    }
}
